lib/spack/spack/binary_distribution.py

```diff
@@ -717,132 +719,24 @@ def _read_specs_and_push_index(
         temp_dir: Location to write index.json and hash for pushing
     """
     for file in file_list:
+        cache_entry: Optional[URLBuildcacheEntry] = None
         try:
-            fetched_spec = spack.spec.Spec.from_dict(read_method(file))
+            cache_entry = read_method(file)
+            spec_dict = cache_entry.fetch_metadata()
+            cache_entry.destroy()
```

Comment:
I wouldn't think you'd need this here since you invoke `destroy()` in the `finally` clause, or am I missing something?